File: src/main/java/marisa/powers/Marisa/PolarisUniquePower.java
package marisa.powers.Marisa;

import com.badlogic.gdx.graphics.Texture;
import com.megacrit.cardcrawl.actions.common.GainEnergyAction;
import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.core.AbstractCreature;
import com.megacrit.cardcrawl.core.CardCrawlGame;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.powers.AbstractPower;

import marisa.Marisa;
import marisa.cards.derivations.GuidingStar;

import com.megacrit.cardcrawl.localization.PowerStrings;

public class PolarisUniquePower extends AbstractPower {

  public static final String POWER_ID = "PolarisUniquePower";
  private static final PowerStrings powerStrings = CardCrawlGame.languagePack
      .getPowerStrings(POWER_ID);
  public static final String NAME = powerStrings.NAME;
  public static final String[] DESCRIPTIONS = powerStrings.DESCRIPTIONS;
  private AbstractPlayer p;
  public boolean Gain;

  public PolarisUniquePower(AbstractCreature owner) {
    Marisa.logger.info("PolarisUniquePower : Init");
    this.name = NAME;
    this.ID = POWER_ID;
    this.type = AbstractPower.PowerType.BUFF;
    updateDescription();
    this.img = new Texture("img/powers/transmute.png");
    this.p = AbstractDungeon.player;
    this.Gain = false;
    this.owner = owner;

    Marisa.logger.info("PolarisUniquePower : Done initing");
  }


  public void stackPower(int stackAmount) {
    Marisa.logger.info("PolarisUniquePower : StackPower");
  }

  public void atStartOfTurnPostDraw() {
    Marisa.logger.info("PolarisUniquePower : Checking");

    for (AbstractCard c : p.drawPile.group) {
      if (c instanceof GuidingStar) {
        this.Gain = true;
      }
    }
    Marisa.logger.info("PolarisUniquePower : Result : " + Gain);
    if (Gain) {
      flash();
      AbstractDungeon.actionManager.addToBottom(new GainEnergyAction(1));
    }
    this.Gain = false;

    Marisa.logger.info("PolarisUniquePower : Done Checking");
  }

  public void updateDescription() {
    Marisa.logger.info("PolarisUniquePower : updating Description");
    this.description = (DESCRIPTIONS[0]);
    Marisa.logger.info("PolarisUniquePower : Done updating Description");
  }
}

File: src/main/kotlin/marisa/powers/Marisa/PolarisUniquePower.kt
package marisa.powers.Marisa

import com.badlogic.gdx.graphics.Texture
import com.megacrit.cardcrawl.actions.common.GainEnergyAction
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.core.AbstractCreature
import com.megacrit.cardcrawl.core.CardCrawlGame
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import com.megacrit.cardcrawl.powers.AbstractPower
import marisa.MarisaContinued
import marisa.cards.derivations.GuidingStar

class PolarisUniquePower(owner: AbstractCreature?) : AbstractPower() {
    private val p: AbstractPlayer = AbstractDungeon.player
    var gain: Boolean = false

    init {
        MarisaContinued.logger.info("PolarisUniquePower : Init")
        name = NAME
        ID = POWER_ID
        type = PowerType.BUFF
        updateDescription()
        img = Texture("img/powers/transmute.png")
        this.owner = owner
        MarisaContinued.logger.info("PolarisUniquePower : Done initing")
    }

    override fun stackPower(stackAmount: Int) {
        MarisaContinued.logger.info("PolarisUniquePower : StackPower")
    }

    override fun atStartOfTurnPostDraw() {
        MarisaContinued.logger.info("PolarisUniquePower : Checking")
        gain = p.drawPile.group.any { it is GuidingStar }
        MarisaContinued.logger.info("PolarisUniquePower : Result : $gain")
        if (gain) {
            flash()
            addToBot(GainEnergyAction(1))
        }
        gain = false
        MarisaContinued.logger.info("PolarisUniquePower : Done Checking")
    }

    override fun updateDescription() {
        MarisaContinued.logger.info("PolarisUniquePower : updating Description")
        description = DESCRIPTIONS[0]
        MarisaContinued.logger.info("PolarisUniquePower : Done updating Description")
    }

    companion object {
        const val POWER_ID = "PolarisUniquePower"
        private val powerStrings = CardCrawlGame.languagePack
            .getPowerStrings(POWER_ID)
        val NAME = powerStrings.NAME
        val DESCRIPTIONS = powerStrings.DESCRIPTIONS
    }
}
